Ignores all errors when using `noqa: X` to only ignore X

When using `# noqa: E501` all errors are excluded.

This is due to the code in https://github.com/PyCQA/flake8-bugbear/blob/6bf40eafe99bb569296c460b691a6c0bebd27ea5/bugbear.py#L39-L40.

`pycodestyle.noqa` searches for r'# no(?:qa|pep8)\b', which matches `# noqa: E501` also.

Without knowing the internals it seems like flake8 itself should include the errors itself - it has logic to ignore only specific errors (https://github.com/PyCQA/flake8/blob/013b925624bd5e4c026e83eb2c6e9f45cafa0dc0/src/flake8/defaults.py#L39-L51).

Example:
```python
s = 'some very, very, very, very, very, very, very, very, very, very, very, very, very, very, very long line'  # noqa: E123
```

flake8-bugbear should report error `t-noqa-colon.py:1:110: B950 line too long (109 > 80 characters)` for it, but is silent.

--- a/bugbear.py
+++ b/bugbear.py
@@ -34,9 +34,6 @@ class BugBearChecker:
         visitor = self.visitor(filename=self.filename, lines=self.lines)
         visitor.visit(self.tree)
         for e in itertools.chain(visitor.errors, self.gen_line_based_checks()):
-            if pycodestyle.noqa(self.lines[e.lineno - 1]):
-                continue
-
             if self.should_warn(e.message[:4]):
                 yield self.adapt_error(e)
 

--- a/tests/b001.py
+++ b/tests/b001.py
@@ -41,13 +41,6 @@ def func(**kwargs):
         # should be except KeyError:
         return
 
-    results = something.call(debug=is_debug)
-    try:
-        results["ok"]
-    except:  # noqa
-        # warning silenced
-        return
-
 
 try:
     pass

--- a/tests/b901.py
+++ b/tests/b901.py
@@ -45,11 +45,6 @@ async def not_broken4():
     return 1
 
 
-def actually_not_broken():
-    yield 2
-    return 1  # noqa
-
-
 def not_broken5():
     def inner():
         return 2

--- a/tests/b950.py
+++ b/tests/b950.py
@@ -4,5 +4,4 @@
 "                              line               is           fine          "
 "                              line               is        still fine          "
 "                              line               is    no longer fine by any measures, yup"
-"                              line               is no longer fine by any measures but is silenced"  # noqa
 "line is fine again"

--- a/tests/test_bugbear.py
+++ b/tests/test_bugbear.py
@@ -51,7 +51,7 @@ class BugbearTestCase(unittest.TestCase):
         expected = self.errors(
             B001(8, 0, vars=("bare `except:`",)),
             B001(40, 4, vars=("bare `except:`",)),
-            B001(54, 0, vars=("`except ():`",)),
+            B001(47, 0, vars=("`except ():`",)),
         )
         self.assertEqual(errors, expected)
 

